src/main/java/frc/robot/subsystems/swervedrive/SwerveSubsystem.java

```diff
@@ -0,0 +1,390 @@
+   * @return Drive command.
+   */
+  public Command driveCommand(DoubleSupplier translationX, DoubleSupplier translationY,
+      DoubleSupplier angularRotationX) {
+    return run(() -> {
+      // Make the robot move
+      swerveDrive.drive(new Translation2d(translationX.getAsDouble() * maximumSpeed, translationY.getAsDouble()),
+          angularRotationX.getAsDouble() * swerveDrive.swerveController.config.maxAngularVelocity,
+          true,
+          false);
+    });
+  }
+
+  /**
+   * Construct the swerve drive.
+   *
+   * @param driveCfg      SwerveDriveConfiguration for the swerve.
+   * @param controllerCfg Swerve Controller.
+   */
+  public SwerveSubsystem(SwerveDriveConfiguration driveCfg, SwerveControllerConfiguration controllerCfg) {
+    swerveDrive = new SwerveDrive(driveCfg, controllerCfg, maximumSpeed);
+  }
+
+  /**
+   * The primary method for controlling the drivebase. Takes a
+   * {@link Translation2d} and a rotation rate, and
+   * calculates and commands module states accordingly. Can use either open-loop
+   * or closed-loop velocity control for
+   * the wheel velocities. Also has field- and robot-relative modes, which affect
+   * how the translation vector is used.
+   *
+   * @param translation   {@link Translation2d} that is the commanded linear
+   *                      velocity of the robot, in meters per
+   *                      second. In robot-relative mode, positive x is torwards
+   *                      the bow (front) and positive y is
+   *                      torwards port (left). In field-relative mode, positive x
+   *                      is away from the alliance wall
+   *                      (field North) and positive y is torwards the left wall
+   *                      when looking through the driver station
+   *                      glass (field West).
+   * @param rotation      Robot angular rate, in radians per second. CCW positive.
+   *                      Unaffected by field/robot
+   *                      relativity.
+   * @param fieldRelative Drive mode. True for field-relative, false for
+   *                      robot-relative.
+   */
+  public void drive(Translation2d translation, double rotation, boolean fieldRelative) {
+    swerveDrive.drive(translation,
+        rotation,
+        fieldRelative,
+        false); // Open loop is disabled since it shouldn't be used most of the time.
+  }
+
+  /**
+   * Drive the robot given a chassis field oriented velocity.
+   *
+   * @param velocity Velocity according to the field.
+   */
+  public void driveFieldOriented(ChassisSpeeds velocity) {
+    swerveDrive.driveFieldOriented(velocity);
+  }
+
+  /**
+   * Drive according to the chassis robot oriented velocity.
+   *
+   * @param velocity Robot oriented {@link ChassisSpeeds}
+   */
+  public void drive(ChassisSpeeds velocity) {
+    swerveDrive.drive(velocity);
+  }
+
+  @Override
+  public void periodic() {
+  }
+
+  @Override
+  public void simulationPeriodic() {
+  }
+
```

Comment:
you can remove these; they have the same implementation in `Subsystem`, the superclass